mmseg/models/losses/cross_entropy_loss.py

```diff
@@ -6,22 +6,58 @@
 from .utils import weight_reduce_loss
 
 
+def calculate_weights(label, num_classes, norm=False, upper_bound=1.0):
+    """Calculate image based classes' weights."""
+    hist = label.float().histc(bins=num_classes, min=0, max=num_classes - 1)
+    hist_norm = hist / hist.sum()
+    if norm:
+        weights = ((hist != 0) * upper_bound * (1 / (hist_norm + 1e-6))) + 1
+    else:
+        weights = ((hist != 0) * upper_bound * (1 - hist_norm)) + 1
+    return weights
+
+
 def cross_entropy(pred,
                   label,
                   weight=None,
+                  img_based_class_weights=None,
+                  batch_weights=True,
```

Comment:
I suggest making `CrossEntropy` simple as it is. 
We may add a new loss module.